code/game/machinery/camera/camera_presets.dm

```diff
@@ -38,21 +38,24 @@
 /obj/machinery/camera/autoname
 	var/number = 0 //camera number in area
 
-//This camera type automatically sets it's name to whatever the area that it's in is called.
+// This camera type automatically sets it's name to whatever the area that it's in is called.
 /obj/machinery/camera/autoname/Initialize(mapload)
-	. = ..()
 	number = 1
 	var/area/A = get_area(src)
-	if(A)
-		for(var/obj/machinery/camera/autoname/C in GLOB.machines)
-			if(C == src)
-				continue
-			var/area/CA = get_area(C)
-			if(CA.type == A.type)
-				if(C.number)
-					number = max(number, C.number + 1)
-		c_tag = "[A.name] #[number]"
+	if(!A)
+		number = rand(1, 100)
+		c_tag = "Unknown #[number]"
+		return ..()
+
+	for(var/obj/machinery/camera/autoname/C in A.contents)
```

Comment:
IIRC looping over an area's contents was super expensive/inefficient. Might be worth benchmarking to see if it's cheaper to loop over the `GLOB.cameranet.cameras` list instead and do an area check

Reply:
Is this true for typecasted loops? I totally forget

Reply:
@AffectedArc07 you might know a bit more about this

Reply:
Its on `Initialize()`. Not particularly an issue.

Coderbus would be best bet for knowledge.

Reply:
Should be fine to go then